rdr_service/tools/tool_libs/genomic_utils.py

```diff
@@ -1582,6 +1584,88 @@ def run_ingestion_for_member_id(self, controller, member_id, bucket_name):
             return 1
 
 
+class CompareIngestionAW2Class(GenomicManifestBase):
+    """
+    Performs a comparison on AW2 file counts and counts in database
+    """
+    def __init__(self, args, gcp_env: GCPEnvConfigObject):
+        super(CompareIngestionAW2Class, self).__init__(args, gcp_env)
+        self.data_objs = []
+
+    def run(self):
+        if self.args.genome_type and self.args.genome_type not in GENOME_TYPE:
+            _logger.error('Valid genome type must be provided - {}'.format(GENOME_TYPE))
+            return 1
+
+        if self.args.gc_site_id and self.args.gc_site_id not in GC_SITE_IDs:
+            _logger.error('Valid gc site id type must be provided - {}'.format(GC_SITE_IDs))
+            return 1
+
+        # Activate the SQL Proxy
+        self.gcp_env.activate_sql_proxy()
+        self.dao = GenomicSetMemberDao()
+
+        _logger.info('Getting file comparison data...')
+        return self.get_file_comparison_data()
+
+    def get_file_comparison_data(self):
+        ignore_flag = 0
+        work_state = 33
+
+        with self.dao.session() as s:
+            records = s.query(
+                functions.count(GenomicSetMember.id).label('rdr_count'),
+                GenomicFileProcessed.filePath
+            ).join(
+                GenomicGCValidationMetrics,
+                GenomicGCValidationMetrics.genomicSetMemberId == GenomicSetMember.id
+            ).join(
+                GenomicFileProcessed,
+                GenomicFileProcessed.id == GenomicGCValidationMetrics.genomicFileProcessedId
+            ).filter(
+                GenomicSetMember.gcSiteId == self.args.gc_site_id,
+                GenomicSetMember.genomeType == self.args.genome_type,
+                GenomicGCValidationMetrics.ignoreFlag == ignore_flag,
+                GenomicWorkflowState != work_state,
+            ).group_by(
+                GenomicFileProcessed.filePath
+            ).all()
+
+            if not records:
+                _logger.info('No records found.')
+                return 1
+
+            _logger.info('{} records found'.format(len(records)))
+
+            for r in records:
+                row_count = self.get_row_count_file(r.filePath)
+                self.data_objs.append({
+                    'rows_in_file': row_count,
+                    'rdr_count': r.rdr_count,
+                    'delta': row_count - r.rdr_count,
+                    'path': r.filePath,
+                })
+            self.output_csv()
+
+    def get_row_count_file(self, file):
+        _logger.info('Getting row count for file...')
+        return sum(1 for l in self.gscp.open(file, 'r'))
```

Comment:
Does this count the header row?

Reply:
Good catch, rectified here: https://github.com/all-of-us/raw-data-repository/pull/2217/commits/14fe29b91a2bb26005111b587f6dfe64aaf2eb21

<img width="1774" alt="Screen Shot 2021-02-08 at 11 28 32 AM" src="https://user-images.githubusercontent.com/5114837/107249691-96a77300-6a01-11eb-83ba-dce360f2559b.png">